common-http/src/test/java/io/cdap/common/http/HttpRequestsTestBase.java

```diff
@@ -110,14 +118,42 @@ public void testHttpStatus() throws Exception {
     testDelete("/api/testDelete", only(200), any(), any(), any());
   }
 
+  private HttpRequest getRequest(HttpRequest.Builder builder, CompletableFuture<String> future) {
+    if (returnResponseStream()) {
+      // Set a small chunk size to verify that we're doing chunked reads correctly.
+      int chunkSize = 5;
+      final String[] streamingResponseBody = {""};
```

Comment:
You can use a `StringBuilder` and just keep appending to it in the onReceived and call toString in onFinish. Seems like it would be simplier.